apps/calendar/src/time/datetime.ts

```diff
@@ -706,6 +708,23 @@ export function getGridLeftAndWidth(
 
     return model;
   });
+
+  const { length } = gridInfo;
+  const gridColWidthMap = fill(length, fill(length, 0));
+  gridInfo.forEach(({ width }, index) => {
+    for (let i = 0; i <= index; i += 1) {
+      for (let j = index; j < length; j += 1) {
+        gridColWidthMap[i][j] += width;
+      }
+    }
+  });
+
+  gridColWidthMap[0][length - 1] = 100;
+
+  return {
+    gridInfo,
+    gridColWidthMap: gridColWidthMap.map((widthList) => widthList.map((width) => toPercent(width))),
```

Comment:
map을 좀 더 줄여볼 수 있겠네요.

```suggestion
    gridColWidthMap: gridColWidthMap.map((widthList) => widthList.map(toPercent)),
```

Reply:
:+1:

Reply:
👍